lib/ao_host.js

```diff
@@ -753,11 +752,11 @@ class AOHost extends AsyncEventEmitter {
    * @private
    */
   async triggerGlobalEvent (section, eventName, ...args) {
-    const instances = Object.values(this.instances)
+    const instances = this.getAOInstances()
 
-    return PI.forEach(instances, async (instance) => (
-      this.triggerAOEvent(instance, section, eventName, ...args)
-    ))
+    await Promise.all(instances.map(async instance => {
+      await this.triggerAOEvent(instance, section, eventName, ...args)
```

Comment:
this one is tricky, some code sections use `await triggerGlobalEvent` and would be delayed by it now

Reply:
https://github.com/bitfinexcom/bfx-hf-algo/blob/4220bed6f6cbb87655fb38781ad743db4a790082/lib/host/ws2/process_message.js#L130-L144

Reply:
> would be delayed by it now

so shouldn't it "await" for `triggerAOEvent` as it was done with PI.forEach before? This can significantly change the behavior of the system, since in the master branch it waits for all events to be sent

Reply:
original code did not await, thats the issue here

```js
    return PI.forEach(instances, async (instance) => (
      this.triggerAOEvent(instance, section, eventName, ...args)
    ))
```

Reply:
```js
async (instance) => ( this.triggerAOEvent(instance, section, eventName, ...args) ))
```

is equal to 
```js
async (instance) => {
  return this.triggerAOEvent(instance, section, eventName, ...args)
 }
```

Reply:
oh its using `()` -- my bad, i missed that